howtos/cmake_install.rst

```diff
@@ -5,33 +5,54 @@ How to reuse cmake install for package() method
 
 It is possible that your project's *CMakeLists.txt* has already defined some
 functionality that extracts the artifacts (headers, libraries, binaries) from
-the build and source folder to a predetermined place.
+the build and source folder to a predetermined place and does the post-processing (*e.g.*, strips rpaths). For example,
+one common practice is to use CMake `install <https://cmake.org/cmake/help/latest/command/install.html>`_ directive
+to that end.
 
-The conan ``package()`` method does exactly that: it defines which files
-have to be copied from the build folder to the package folder.
+When using conan, you need to wrap that functionality in the conan ``package()`` method.
 
-If you want to reuse that functionality, you can do it with cmake.
-
-Invoke cmake with ``CMAKE_INSTALL_PREFIX`` using the ``package_folder`` variable.
-If the ``cmake install`` target correctly copies all the required libraries, headers, etc. to the ``package_folder``,
-then the ``package()`` method is not required.
+The following excerpt shows how to build and package with CMake within conan. Mind that you need to configure CMake
+both in ``build()`` and in ``package()`` since these methods are called independently.
 
 .. code-block:: python
 
-    def build(self):
+    def configure_cmake(self):
         cmake = CMake(self)
-        cmake.configure()
+        cmake.definitions["SOME_DEFINITION"] = "On"
+
+        return cmake
+
+    def build(self):
+        cmake = self.configure_cmake()
         cmake.build()
-        cmake.install()
-        # equivalent to
-        # args += ['-DCMAKE_INSTALL_PREFIX="%s"' % self.package_folder]
-        # self.run('cmake "%s/src" %s %s'
-        #          % (self.source_folder, cmake.command_line, ' '.join( args ) ) )
-        # self.run("cmake --build . --target install %s" % cmake.build_config)
 
     def package(self):
-        # nothing to do here now
+        cmake = self.configure_cmake()
+        cmake.install()
+
+If you want to specify a different ``CMAKE_INSTALL_PREFIX`` for your package, you need to specify it as
```

Comment:
Dont see the point here to change CMAKE_INSTALL_PREFIX as you want it to be pointing to the package folder as it is already. Cant see the use case

Reply:
Please mind that we don't want to change `CMAKE_INSTALL_PREFIX`. We want to change the default `package_folder` to an arbitrary folder. Conan will pass this value to CMake which eventually sets `CMAKE_INSTALL_PREFIX`. 

We have a release script which we use to build and install in arbitrary folders. We use conan only to manage the dependencies, but don't have any downstream conan consumers.

I wanted to include this example to help other people as it took us long to figure it out, but I suppose it makes most sense if we just keep using it internally and leave it undocumented here as long as API is not stable.